### PR TITLE
217 feature split build workflows into smaller ones

### PR DESCRIPTION
**Pull request by leon1995**
## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have updated the [changelog](../CHANGELOG.md) accordingly.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: .github/workflows/mypy.yml [hunk 1]
```diff
@@ -25,4 +25,5 @@ jobs:
         with:
           checkName: 'mypy'   # NOTE: this needs to be the same as the job name
         env:
-          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
\ No newline at end of file
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        continue-on-error: true # https://github.com/Draegerwerk/sdc11073/issues/156
```
Comment (maximilianpilz) [inline .github/workflows/mypy.yml:29]: remove this, as it was already done in https://github.com/Draegerwerk/sdc11073/pull/216 when this PR was opened

File: .github/workflows/ruff.yml [hunk 2]
```diff
@@ -10,4 +10,5 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v3
-      - uses: chartboost/ruff-action@v1
\ No newline at end of file
+      - uses: chartboost/ruff-action@v1
+        continue-on-error: true # https://github.com/Draegerwerk/sdc11073/issues/156
```
Comment (maximilianpilz) [inline .github/workflows/ruff.yml:14]: remove this, as it was already done in https://github.com/Draegerwerk/sdc11073/pull/216 when this PR was opened

File: .github/workflows/build.yml
Comment (maximilianpilz) [inline .github/workflows/build.yml:1]: add a newline at the end of this file

**Review by maximilianpilz — CHANGES_REQUESTED** [review]

File: .github/workflows/mypy.yml (hunk 1 above)
Comment (leon1995) [reply .github/workflows/mypy.yml]: done

File: .github/workflows/ruff.yml (hunk 2 above)
Comment (leon1995) [reply .github/workflows/ruff.yml]: done

File: .github/workflows/build.yml
Comment (leon1995) [reply .github/workflows/build.yml:1]: done

File: pyproject.toml [hunk 3]
```diff
@@ -77,6 +77,17 @@ path = "src/sdc11073/__init__.py"
 [tool.pytest.ini_options]
 testpaths = "tests/"
 
+# https://docs.pytest.org/en/6.2.x/logging.html#live-logs
```
Comment (maximilianpilz) [inline pyproject.toml:80]:
```suggestion
# https://docs.pytest.org/en/latest/how-to/logging.html#live-logs
```

**Review by maximilianpilz — CHANGES_REQUESTED** [review]

**Review by maximilianpilz — APPROVED** [review]

**Pull request merged**